File: backend/B2BApplication/DTO/CatalogueKeyPair.cs
namespace B2BApplication.DTO
{
    public class CatalogueKeyPair
    {
        public int id { get; set; }
        public string name { get; set; }
        //public string? parent { get; set; }
        public int? parentId { get; set; }

       public int niveau { get; set; }
        public int nbArticle { get; set; }
        public List<CatalogueKeyPair> value { get; set; } = null;
         
    }
}

File: frontend/src/app/models/catalogueKeyPair.ts
export class CatalogueKeyPair
{
    id:number
    value:CatalogueKeyPair[]
    name:string
    nbArticle:number
    constructor( id:number,value:CatalogueKeyPair[],name:string){
        this.id=id
        this.value=value
        this.name=name
        this.nbArticle=0
    }
}